pkg/test/framework/components/echo/kube/workload_manager.go

```diff
@@ -213,9 +213,6 @@ func (m *workloadManager) onPodAddOrUpdate(pod *kubeCore.Pod) error {
 			if newWorkload.IsReady() {
 				workloadReady = newWorkload
 			}
-
-			// Now add the previous workload.
-			newWorkloads = append(newWorkloads, workload)
```

Comment:
I'm not sure if the order would matter, but assuming that it does, would a better fix be slicing the list at `i` (from the loop) and inserting there? 

I'll let Nate look.